master/master.cfg

```diff
@@ -154,22 +191,26 @@ build_factory.addStep(Git(repourl=spl_repo, workdir="build/spl",
     mode="full", method="clobber", codebase="spl",
     logEnviron=False, getDescription=True,
     lazylogfiles=True,
-    description=["cloning"], descriptionDone=["cloned"]))
+    description=["cloning"], descriptionDone=["cloned"],
+    doStepIf = do_step_build_spl,
+    hideStepIf=lambda results, s: results==SKIPPED))
 build_factory.addStep(Git(repourl=zfs_repo, workdir="build/zfs",
     mode="full", method="clobber", codebase="zfs",
     logEnviron=False, getDescription=True,
     lazylogfiles=True,
-    description=["cloning"], descriptionDone=["cloned"]))
+    description=["cloning"], descriptionDone=["cloned"],
+    doStepIf = do_step_build_zfs,
+    hideStepIf=lambda results, s: results==SKIPPED))
 build_factory.addStep(Git(repourl=lustre_repo, workdir="build/lustre",
     mode="full", method="clobber", codebase="lustre",
     logEnviron=False, getDescription=True,
     lazylogfiles=True,
+    description=["cloning"], descriptionDone=["cloned"],
     doStepIf = do_step_build_lustre,
-    hideStepIf=lambda results, s: results==SKIPPED,
-    description=["cloning"], descriptionDone=["cloned"]))
+    hideStepIf=lambda results, s: results==SKIPPED))
 build_factory.addStep(ShellCommand(
     workdir="build/linux", env={'PATH' : bin_path,
-        'LINUX_BUILTIN' : util.Interpolate('%(prop:builtin:-no)s') },
+        'LINUX_BUILTIN' : util.Interpolate('%(kw:p)s', p=get_builtin) },
     command=["runurl", bb_url + "bb-build-linux.sh"],
```

Comment:
It looks like `bb-build-linux.sh` is also going to need an update such that it doesn't always assume the `../spl/` directory exists.